src/main/kotlin/org/jitsi/jicofo/bridge/ExternalBridgeSelectionStrategy.kt

```diff
@@ -0,0 +1,110 @@
+        bridges: MutableList<Bridge>?,
+        conferenceBridges: MutableMap<Bridge, Int>?,
+        participantRegion: String?
+    ): Bridge {
+        val url = ExternalBridgeSelectionStrategyConfig.config.url
+            ?: throw Exception("ExternalBridgeSelectionStrategy requires url to be provided")
+
+        val requestBody = JSONObject()
+        requestBody["bridges"] = bridges?.map {
+            mapOf(
+                "jid" to it.jid.toString(),
+                "version" to it.version,
+                "colibri2" to it.supportsColibri2(),
+                "relay_id" to it.relayId,
+                "region" to it.region,
+                "stress" to it.stress,
+                "operational" to it.isOperational,
+                "graceful_shutdown" to it.isInGracefulShutdown,
+                "draining" to it.isDraining,
+            )
+        }
+        requestBody["conference_bridges"] = conferenceBridges?.mapKeys { it.key.jid.toString() }
+        requestBody["participant_region"] = participantRegion
+        requestBody["fallback_strategy"] = ExternalBridgeSelectionStrategyConfig.config.fallbackStrategy
+
+        val request = HttpRequest
+            .newBuilder()
+            .POST(HttpRequest.BodyPublishers.ofString(requestBody.toJSONString()))
+            .uri(URI.create(url))
+            .headers("Content-Type", "application/json")
+            .timeout(ExternalBridgeSelectionStrategyConfig.config.timeout)
+            .build()
+
+        val response: HttpResponse<String>
+        try {
+            response = httpClient.send(request, HttpResponse.BodyHandlers.ofString())
+        } catch (exc: Exception) {
+            logger.error("ExternalBridgeSelectionStrategy: HTTP request failed with ${exc}, using fallback strategy")
+            return fallback(bridges, conferenceBridges, participantRegion)
+        }
+
+        val statusCode = response.statusCode()
+        if (statusCode !in 200..299) {
+            logger.error("ExternalBridgeSelectionStrategy: HTTP request failed with ${statusCode}, using fallback strategy")
+            return fallback(bridges, conferenceBridges, participantRegion)
+        }
+
+        val responseBody: JSONObject
+        try {
+            responseBody = JSONValue.parseWithException(response.body()) as JSONObject
+        } catch (exc: Exception) {
+            logger.error("ExternalBridgeSelectionStrategy: HTTP response parsing failed with ${exc}, using fallback strategy")
+            return fallback(bridges, conferenceBridges, participantRegion)
+        }
+
+        val selectedBridgeIndex = responseBody["selected_bridge_index"] as? Number
+
+        if (selectedBridgeIndex == null) {
+            logger.error("ExternalBridgeSelectionStrategy: HTTP response selected_bridge_index missing or invalid, using fallback strategy")
+            return fallback(bridges, conferenceBridges, participantRegion)
+        }
+
+        val bridge = bridges!![selectedBridgeIndex.toInt()]
```

Comment:
What do you think about having the external service return an ID (JID) of the selected bridge instead of an index? We want to have the service return a bridge that jicofo doesn't yet know about (this will require more changes, but it will be easier to have the HTTP API ready to support it and not have to change).

Reply:
Yes, this seems sensible. Interesting use case to use a bridge it doesn't know about yet. Launching bridges on-demand?